tests/test_tokenization_fast.py

```diff
@@ -21,117 +24,92 @@
 from transformers.tokenization_transfo_xl import TransfoXLTokenizerFast
 
 
-class FastTokenizerMatchingTest(unittest.TestCase):
-    def setUp(self) -> None:
-        with open("tests/fixtures/sample_text.txt") as f_data:
-            self._data = f_data.read().replace("\n\n", "\n").strip()
-
-    def assert_sequence_almost_equals(self, a, b, threshold):
-
-        # Handle padding
-        if len(a) != len(b):
-            max_len = max(len(a), len(b))
+Tokenizer = namedtuple("Tokenizer", ["name", "rust", "python", "vocab_key"])
 
-            # Pad with a negative number as vocab doesnt allow idx < 0
-            # if will be tracked as differences
-            if len(a) < max_len:
-                a += [-1] * (max_len - len(a))
 
-            if len(b) < max_len:
-                b += [-1] * (max_len - len(b))
+class CommonFastTokenizerTest(ABC, unittest.TestCase):
 
-        # Convert to numpy for convenience
-        a_, b_ = np.array(a), np.array(b)
+    TOKENIZERS_CLASSES = frozenset([
+        Tokenizer("Bert", BertTokenizerFast, BertTokenizer, "vocab_file"),
+        Tokenizer("DistilBert", DistilBertTokenizerFast, DistilBertTokenizer, "vocab_file"),
+        Tokenizer("Roberta", RobertaTokenizerFast, RobertaTokenizer, "vocab_file"),
+    ])
 
-        # Compute elementwise difference
-        inputs_diffs = a_ - b_
-        inputs_diff = np.count_nonzero(inputs_diffs)
-        self.assertLessEqual(inputs_diff / a_.shape[0], threshold)
+    def setUp(self) -> None:
+        with open("tests/fixtures/sample_text.txt", encoding="utf-8") as f_data:
+            self._data = f_data.read().replace("\n\n", "\n").strip()
 
-    def assert_tokenization_python_rust_almost_equals(self, tokenizer_p, tokenizer_r, threshold: float):
+    def test_fast_align_python(self):
+        for (name, rust_cls, python_cls, vocab_key) in self.TOKENIZERS_CLASSES:
+            for pretrained_name in python_cls.pretrained_vocab_files_map[vocab_key].keys():
+                with self.subTest("{} ({})".format(name, pretrained_name)):
```

Comment:
This seems to me like to optimal organisation. Testing on every checkpoint on every tokenizer organized in subTests is really thorough.

Does it take a while? Should we mark this as slow or is it fast enough?

Reply:
It takes around 1 min for doing everything